# Logistic Regression

Sentiment classification on the IMDB Dataset using logistic regression model to predict positive or negative reviews.

In [1]:
import numpy as np
from tensorflow.keras.datasets import imdb
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, classification_report

17464789/17464789 ━━━━━━━━━━━━━━━━━━━━ 1s 0us/step


In [6]:
(x_train, y_train), (x_test, y_test) = imdb.load_data(num_words=10000)

def vectorize_sequences(sequences, dimension=10000):
    results = np.zeros((len(sequences), dimension))
    for i, seq in enumerate(sequences):
        results[i, seq] = 1
    return results

x_train_vec = vectorize_sequences(x_train)
x_test_vec = vectorize_sequences(x_test)

In [7]:
lr_model = LogisticRegression(max_iter=1000)
lr_model.fit(x_train_vec, y_train)
y_pred_lr = lr_model.predict(x_test_vec)

In [8]:
print("Logistic Regression Accuracy:", accuracy_score(y_test, y_pred_lr))
print(classification_report(y_test, y_pred_lr))

Logistic Regression Accuracy: 0.86128
              precision    recall  f1-score   support

           0       0.86      0.86      0.86     12500
           1       0.86      0.86      0.86     12500

    accuracy                           0.86     25000
   macro avg       0.86      0.86      0.86     25000
weighted avg       0.86      0.86      0.86     25000



The model achieves about 86.13% accuracy, with balanced precision, recall, and F1-score of 0.86 for both classes. This indicates stable performance with no strong bias toward either positive or negative reviews.